### PR TITLE
Added COPYRIGHT notice

### PR DESCRIPTION
**Pull request by bvssvni**
Closes https://github.com/PistonDevelopers/piston/issues/1054

cc @PistonDevelopers/pistoncollaborator @PistonDevelopers/admins @PistonDevelopers/owners 

This PR adds special legal terms and conditions for license change. These rules are designed for prevent unavoidable obstacles in the long term with a project such as Piston, where there are many contributors, by applying a small change to how votes are counted in the decision process.

Would like to hear what you think!

### CODE REVIEW

File: COPYRIGHT.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+Default votes apply for contributions before this date.
+
+Here is a diagram explaining when copyright holder is for person or group:
+
+               _____________________________________________________________
+              | 100% agreement | > 95% agreement | > 3 months               |
+              |----------------|-----------------|--------------------------|
+default       | 1 per person   | 1 per person    | 1 per person             |
+              |----------------|-----------------|--------------------------|
+lend vote     | 1 per person   | 1 for group     | if 100% agreement:       |
```
Comment (panicbit) [inline COPYRIGHT.md]: The table seems to get messed up here

File: COPYRIGHT.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+### PistonDevelopers COPYRIGHT Notice
+
+By contributing to this project after
+
+    29-05-2016
```
Comment (natemara) [inline COPYRIGHT.md]: I would recommend an ISO-8061 date here. 2016-05-29

Comment (bvssvni) [inline COPYRIGHT.md]: Thanks!

File: COPYRIGHT.md (hunk 1 above)
Comment (bvssvni) [inline COPYRIGHT.md]: I've made it a bit shorter. Does it work?

Comment (panicbit) [inline COPYRIGHT.md]: Nope, still borked. I think `default` and `lend vote` in front of the table mess it up.

**Comment by bvssvni** [comment]
@panicbit I forgot to add back-quotes for putting the table in a code block. Now it should work.

**Comment by bvssvni** [comment]
OK, we could try this for period and see how people responds. If there are any problems we can just change it back. We still have to do transition to dual licensing between Apache 2 and MIT the normal way, so these rules won't have any effect before potential new licenses, which probably will take many years and give us time to evaluate this thoroughly. Could also try a simpler version in new projects and see how that goes.

Merging.

**Comment by nox** [comment]
What does "Default votes apply for contributions before this date." mean? What does it mean for anyone who contributed before the date in that weird agreement? AFAICT if you ever wish to change the license of piston, you will need to contact these contributors, whatever the agreement says, or rewrite the code they authored.

**Comment by bvssvni** [comment]
@nox "Default votes apply for contributions before this date." means that nothing changes for anyone who contributed before the date, just like you said. I can fix it to make it clearer.

They will have to be contacted as normal. It is the same for newer contributions too, except those agree to lend their vote to a majority of 95% in case they do not respond within 3 months.

**Comment by metajack** [comment]
This is a very odd thing, and I'm curious if you involved legal counsel in creating this.

**Comment by cmr** [comment]
Yeah, this is pretty weird, and I don't like it.

**Comment by adambadawy** [comment]
> These rules are designed for prevent unavoidable obstacles in the long term with a project such as Piston, where there are many contributors, by applying a small change to how votes are counted in the decision process.

Please clarify. I thought majority rules in a vote. So what does this do? Could you give an example of an unavoidable obstacle that this prevents?

**Comment by adambadawy** [comment]
I didn't see this: https://github.com/PistonDevelopers/piston/pull/1082

I'll leave it here for others.

**Comment by bvssvni** [comment]
OK, this blew up, and I'll remove it. People got scared and felt this was weird. It is clear that the COPYRIGHT notice doesn't communicate clearly what it was supposed to do, so there is no worth in keeping it. Sorry for disappointing you.

@metajack There was no legal counseling involved. I wish I had an expert available on such things, because this is tricky to do right, and making the rules unambiguous is not enough, it has to sound comforting to make people pleased. I will try find somebody to help next time I do something like this.

Thanks for the feedback anyway! I will learn from this such that next time I'm doing something like this, I will think twice and do everything I can to make sure it won't scare anybody.

**Comment by flying-sheep** [comment]
IANAL but actually I think it would have been ineffective anyway.

all projects with a CLA have you sign it explicitly beforehand. Therefore I think this may be necessary for a CLA to be effective.

**Pull request merged**